coding-standard/action.yml

```diff
@@ -0,0 +1,138 @@
+
+  version:
+    type: string
+    required: false
+    default: "31"
+    description: "The version of the coding standard to use. If not provided, will use the latest version."
+
+  severity:
+    type: string
+    required: false
+    default: "5"
+    description: "The minimum severity required to display an error or warning (default: 5)"
+
+  warning_severity:
+    type: string
+    required: false
+    default: "5"
+    description: "The minimum severity required to display a warning"
+
+  error_severity:
+    type: string
+    required: false
+    default: "5"
+    description: "The minimum severity required to display an error"
+
+  baseline_version:
+    type: string
+    required: false
+    default: "1.1.2"
+    description: "The version of the phpcs baseline to use. if not required it uses default one"
+
+runs:
+  using: composite
+  steps:
+    - name: Checkout Project
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        path: standard
+
+    - name: Set PHP Version
+      uses: shivammathur/setup-php@v2
+      with:
+        php-version: ${{ inputs.php_version }}
+        tools: composer:v${{ inputs.composer_version }}
+        coverage: none
+
+    - name: Install Coding Standard && Codesniffer baseline
+      shell: bash
+      working-directory: standard
+      run: |
+        git config --global advice.detachedHead false
+        composer require "magento/magento-coding-standard:${{ inputs.version || '*' }}"
+        composer config --no-plugins allow-plugins.digitalrevolution/php-codesniffer-baseline true
+        composer require --dev "digitalrevolution/php-codesniffer-baseline:${{ inputs.baseline_version }}"
+
+    - name: Register Coding Standard
+      shell: bash
+      working-directory: standard
+      run: vendor/bin/phpcs --config-set installed_paths ${{ github.workspace }}/standard/vendor/magento/magento-coding-standard,${{ github.workspace }}/standard/vendor/phpcompatibility/php-compatibility
+
+    - name: Get all changed files
+      id: changed-files
+      uses: tj-actions/changed-files@v35
+
+    - name: Checkout - Before Merge
+      shell: bash
+      working-directory: standard
+      id: previous-commit-files
+      run: |
+        if ${{ github.event_name == 'pull_request' }}; then
+          git checkout ${{ github.event.pull_request.base.ref }}
+        else
+          git checkout ${{ github.event.before }}
+        fi
+
+        for file in ${{ steps.changed-files.outputs.all_modified_files }}; do
+          echo $file >> ${{ github.workspace }}/file_lists.txt
+        done
+        cat ${{ github.workspace }}/file_lists.txt
+
+    - name: Filter php files and execute phpcs - Before Merge
+      shell: bash
+      id: phpcs_before
+      working-directory: standard
+      run: |
+        grep -i "\.php$" ${{ github.workspace }}/file_lists.txt > ${{ github.workspace }}/phpcs_files.txt
```

Comment:
The two `grep` commands could probably be unified with something like `php -iE "\.php|\.phtml"`.

Reply:
Thank you, need to upgrade on Linux commands